File: BookLocal.Data/Models/MainCategory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookLocal.Data.Models
{
    public class MainCategory
    {
        [Key]
        public int MainCategoryId { get; set; }

        [Required]
        [MaxLength(100)]
        public required string Name { get; set; }

        public virtual ICollection<ServiceCategory> ServiceCategories { get; set; } = new List<ServiceCategory>();
    }
}

File: booklocal_mobile/lib/core/models/main_category.dart
class MainCategory {
  final int mainCategoryId;
  final String name;

  MainCategory({
    required this.mainCategoryId,
    required this.name,
  });

  factory MainCategory.fromJson(Map<String, dynamic> json) {
    return MainCategory(
      mainCategoryId: json['mainCategoryId'] ?? 0,
      name: json['name'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'mainCategoryId': mainCategoryId,
      'name': name,
    };
  }
}
